src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/simple/DPBasedBushyJoinPlanOptimizer.java

```diff
@@ -0,0 +1,47 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.poptimizer.simple;
+
+import se.liu.ida.hefquin.engine.queryproc.impl.poptimizer.QueryOptimizationContext;
+import se.liu.ida.hefquin.engine.utils.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DPBasedBushyJoinPlanOptimizer extends DPBasedJoinPlanOptimizer {
+
+    public DPBasedBushyJoinPlanOptimizer(final QueryOptimizationContext ctxt ) {
+        super(ctxt);
+    }
+
+    @Override
+    public <T> List<Pair<List<T>, List<T>>> splitIntoSubSets( final List<T> superset ) {
```

Comment:
Can you please create a corresponding unit test that checks the correct implementation of this function. Testing with a `List<Integer>` containing three elements, 1-3, may be enough. In this unit test, the checking of the resulting list of pairs of lists should be done not only based on the size of that resulting list but by actually checking that all expected pairs are indeed contained in the resulting list.